prow/tide/tide.go

```diff
@@ -1153,6 +1158,12 @@ func (c *changedFilesAgent) prune() {
 	c.nextChangeCache = make(map[changeCacheKey][]string)
 }
 
+func refGetterFactory(ref string) config.RefGetter {
+	return func() (string, error) {
+		return ref, nil
+	}
+}
+
 func (c *Controller) presubmitsByPull(sp *subpool) (map[int][]config.Presubmit, error) {
 	presubmits := make(map[int][]config.Presubmit, len(sp.prs))
```

Comment:
As mentioned [here](https://github.com/kubernetes/test-infra/pull/13644#discussion_r310824633) this won't be sufficient for batches. We can't just union the presubmits required by each PR in the batch which is what we are doing now since this PR doesn't update `accumulateBatch` or `trigger`.